libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteProgressChangeListener.kt

```diff
@@ -69,16 +68,9 @@ internal class MapRouteProgressChangeListener(
                     val dist = routeProgress.distanceTraveled() / totalDist
                     if (dist > 0) {
                         val deferredExpression = async(Dispatchers.Default) {
-                            val lineString: LineString =
-                                routeLine.getLineStringForRoute(currentRoute!!)
-                            buildRouteLineExpression(
-                                currentRoute,
-                                lineString,
-                                true,
-                                dist.toDouble(),
-                                routeLine::getRouteColorForCongestion
-                            )
+                            routeLine.getExpressionAtOffset(dist)
                         }
+                        delay(600)
```

Comment:
Why this magic number `600` 🤔?

Reply:
The route progress occurs about every second. The delay allows the puck to move a bit before updating the vanishing point of the line.  That delay value looked the best when I was testing but it could be increased or decreased.